crates/nargo_cli/tests/test_data/brillig_nested_slices/src/main.nr

```diff
@@ -0,0 +1,59 @@
+use dep::std::slice;
+use dep::std;
+
+// Tests nested slice passing to/from functions
+unconstrained fn push_back_to_slice<T>(slice: [T], item: T) -> [T] {
+    slice.push_back(item)
+}
+
+struct NestedSliceStruct {
+    id: Field,
+    arr: [Field]
+}
+
+unconstrained fn create_foo(id: Field, value: Field) -> NestedSliceStruct {
+    let mut arr = [id];
+    arr = arr.push_back(value);
+    NestedSliceStruct { id, arr }
+}
+
+unconstrained fn main(a: Field, b: Field) {
+    let mut slice = [create_foo(a, b), create_foo(b, a)];
+    assert(slice.len() == 2);
+
+    assert(slice[0].id == a);
+    assert(slice[0].arr[0] == a);
+    assert(slice[1].id == b);
+    assert(slice[1].arr[1] == a);
+
+    slice = push_back_to_slice(slice, create_foo(0, 42));
```

Comment:
Could you also test the rest of the slice builtins such as `pop_back`, `pop_front`, and `push_front`?